Lib/test/datetimetester.py

```diff
@@ -2535,6 +2535,12 @@ def test_strptime(self):
         with self.assertRaises(ValueError): strptime("-2400", "%z")
         with self.assertRaises(ValueError): strptime("-000", "%z")
 
+    def test_strptime_ampm(self):
+        for hour in range(0, 24):
+            with self.subTest(hour=hour):
+                hour_date = (1999, 3, 17, hour, 44, 55, 2, 76, 0)
+                self.assertEqual(self.theclass.strptime(_time.strftime("%I %p", hour_date), "%I %p").hour, hour)
```

Comment:
This line is I think too long for PEP8. If I may suggest a refactoring for this function that I think makes it a bit cleaner:

```python
    def test_strptime_ampm(self):
        dt = datetime(1999, 3, 17, 0, 44, 55, 2, 76, 0)
        for hour in range(0, 24):
            with self.subTest(hour=hour):
                new_dt = dt.replace(hour=hour)
                dt_str = new_dt.strftime("%I %p")

                self.assertEqual(self.theclass.strptime(dt_str, "%I %p").hour,
                                 hour)
```

Reply:
I likely opted to ignore line length, since much of the rest of the file has PEP8 issues, but I can change that.

Out of curiosity, why do you feel that using `dt.replace` is cleaner than generating a new datetime value directly each iteration?